### DISCUSSION TITLE
Running Sensor Tasks in Series not Parallel

### DISCUSSION THREAD

**Discussion by PascalStehling** [dagster-io/dagster#8313] [Q&A] [answered]
Hi,

lets say I have an sensor which runs an Job on each file in a Folder and checks if new file is added to the Folder. These Files are named in Order (1,2,3,4, ...) and each new File Name is higher than the one before. The Data of each file is dependent on the previous file and the Data will be stored in a Database.

Is there now a way to run the Sensor so, that if multiple files are added at once (eg. 7,8,9) that first the job for 7 runs, after that 8 and than 9 and NOT all 3 Jobs in Parallel? Preferably this should work with the DockerRunLauncher.

Thanks in advance :)

**Comment by johannkm** [comment, marked as answer]
The simplest solution would be to use run concurrency limits. You could set a limit of 1 for a specific tag key, and the runs will execute serially in the order that your sensor submitted them. This also could be brittle- if one run fails, Dagster doesn't know about the dependency and will continue to launch the others. You may be able to mitigate this with op retries
